src/Microsoft.Extensions.Primitives/StringTokenizer.cs

```diff
@@ -31,21 +43,21 @@ public StringTokenizer(string value, char[] separators) : this((StringSegment)va
         /// <param name="separators">The characters to tokenize by.</param>
         public StringTokenizer(StringSegment value, char[] separators)
         {
-            if (value == null)
+            if (!value.HasValue)
```

Comment:
Currently converts null into a `new StringValues(null)` then does `==(StringValues, StringValues)` on it